src/compiler/binder.ts

```diff
@@ -2656,7 +2666,7 @@ namespace ts {
 
             // If a FunctionDeclaration is async, then it is TypeScript syntax.
             if (modifierFlags & ModifierFlags.Async) {
-                transformFlags |= TransformFlags.AssertTypeScript;
+                transformFlags |= TransformFlags.AssertTypeScript | TransformFlags.AssertES2017;

```

Comment:
According to [this](https://github.com/tc39/proposals/blob/master/finished-proposals.md) async/await is finished and will be part of es2017, so I believe you have a valid point there.